hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/datanode/DecommissionStatusSubCommand.java

```diff
@@ -134,9 +179,62 @@ private void printCounts(DatanodeDetails datanode, JsonNode counts, int numDecom
           return;
         }
       }
-      System.err.println("Error getting pipeline and container counts for " + datanode.getHostName());
-    } catch (NullPointerException ex) {
-      System.err.println("Error getting pipeline and container counts for " + datanode.getHostName());
+      System.err.println(errMsg);
+    } catch (IOException e) {
+      System.err.println(errMsg);
     }
   }
+
+  private Map<String, Object> getDatanodeDetails(DatanodeDetails datanode) {
+    Map<String, Object> detailsMap = new LinkedHashMap<>();
+    detailsMap.put("uuid", datanode.getUuid().toString());
+    detailsMap.put("networkLocation", datanode.getNetworkLocation());
+    detailsMap.put("ipAddress", datanode.getIpAddress());
+    detailsMap.put("hostname", datanode.getHostName());
+    return detailsMap;
+  }
+
+  private Map<String, Object> getCounts(DatanodeDetails datanode, JsonNode counts, int numDecomNodes) {
```

Comment:
There is quite a bit of code duplication between `printCounts` and `getCounts`. To avoid that, we can use this `getCounts` method in `printCounts` and then display it appropriately. Something like this:
```
  private void printCounts(DatanodeDetails datanode, JsonNode counts, int numDecomNodes) {
    Map<String, Object> countsMap = getCounts(datanode, counts, numDecomNodes);
    System.out.println("Decommission Started At : " + countsMap.get("decommissionStartTime"));
    System.out.println("No. of Unclosed Pipelines: " + countsMap.get("numOfUnclosedPipelines"));
    System.out.println("No. of UnderReplicated Containers: " + countsMap.get("numOfUnderReplicatedContainers"));
    System.out.println("No. of Unclosed Containers: " + countsMap.get("numOfUnclosedContainers"));
  }
```